--- a/src/interactive_marker_client.cpp
+++ b/src/interactive_marker_client.cpp
@@ -76,7 +76,7 @@
       graph_interface_,
       services_interface_,
       topic_namespace_ + "/get_interactive_markers",
-      rclcpp::ServicesQoS(),
+      rclcpp::ServicesQoS().get_rmw_qos_profile(),
       nullptr);
 
     feedback_pub_ = rclcpp::create_publisher<visualization_msgs::msg::InteractiveMarkerFeedback>(
